### PR TITLE
Add comment to Dockerfile, use specific tag for FROM

### PR DESCRIPTION
**Pull request by tbugfinder**
Referring to #327 a comment is added to Dockerfiles and the latest v3.11 tag is applied for builds using FROM docker hub.
During test-build scripts were not executed, therefore executable bit was set explicitly.

### CODE REVIEW

File: jenkins/master/Dockerfile [hunk 1]
```diff
@@ -11,16 +12,19 @@ RUN yum -y install openssl gnutls-utils \
 
 # fetch certificates and store them in tmp directory
 COPY ./import_certs.sh /usr/local/bin/import_certs.sh
-RUN import_certs.sh
+RUN chmod o+x /usr/local/bin/import_certs.sh \
```
Comment (gerardcl) [inline jenkins/master/Dockerfile]: hi! have you tried `sh import_certs.sh` instead of having to change permission set?

**Comment by gerardcl** [comment]
when moving files around permissions might change (and, I agree, fixing the permission set before executing is an option), but it is recommended to stick to execution permission at user level as much as possible via `sh script.sh` instead of fixing, for example, execution rights forever.

File: jenkins/master/Dockerfile (hunk 1 above)
Comment (tbugfinder) [reply jenkins/master/Dockerfile]: I don't have a preference, yet.
If permissions are not set as needed, those might be wrong any way.
This also means that `sh import_certs.sh` could execute successfully, but permissions of `import_certs.sh` might not be as expected.

**Comment by tbugfinder** [comment]
My understanding is that build should pull latest image of `openshift/jenkins-slave-base-centos7`.  Even if ODS is running on OCP3.6, the image tagged `:latest` would be pulled.
The issue in this case is, that tag `:latest` doesn't refer to latest image as it is not properly maintained (https://github.com/openshift/jenkins) and is being migrated to [quai.io](https://quay.io/repository/openshift/origin-jenkins?tag=latest&tab=tags) without support of OCP3.x.

**Comment by michaelsauter** [comment]
@tbugfinder Then maybe best it to make it configurable ... that we show that you should set this to a tag corresponding to the OpenShift version used.

Regarding file permissions, I prefer to fix the bits in the Dockerfile. We have the control there, so I think this is better ... but hey, both works.

**Comment by tbugfinder** [comment]
@michaelsauter How about `ORIGIN_IMAGE_TAG=latest|v3.11|4.4` (derived from ODS_IMAGE_TAG) ?

**Comment by michaelsauter** [comment]
@tbugfinder Could work. I am not sure whether you would need different tags based on which image you use?! Also, maybe we want to allow configuring the whole image, not just the tag?

File: jenkins/slave-base/Dockerfile.centos7 [hunk 2]
```diff
@@ -1,4 +1,7 @@
-FROM openshift/jenkins-slave-base-centos7
+# BuildConfig (../ocp-config/bc.yml) overrides FROM instruction
```
Comment (michaelsauter) [inline jenkins/slave-base/Dockerfile.centos7]: Shouldn't we add the config there then?

Comment (tbugfinder) [reply jenkins/slave-base/Dockerfile.centos7]: yes, partially it's already there.
https://github.com/opendevstack/ods-core/blob/master/jenkins/ocp-config/bc.yml#L121-L124

Comment (michaelsauter) [reply jenkins/slave-base/Dockerfile.centos7]: Yes sorry, I should have commented for the master image. The slave image config is already there. I would have actually preferred if your proposed solution would also be just one variable for both image and tag - I don't see the value of having both?

**Comment by hugowschneider** [comment]
Hi guys, I just notice the tests picked up the first PR error:

```
Cloning "http://172.17.0.1:8080/scm/opendevstack/ods-core.git" ...
	Commit:	53bfca04109115fde9ba8d32b48980e197cc6e0b (Merge 017e9801635268c47af53ed449ae0887ad6831e8 into c89cc4c33ac466ed848b390cfa..)
	Author:	tbugfinder <github@online.ms>
	Date:	Sun Jan 12 20:39:28 2020 +0000
Replaced Dockerfile FROM image ${JENKINS_AGENT_BASE_FROM_IMAGE}

Pulling image openshift/jenkins-slave-base-centos7 ...
Pulled 1/7 layers, 17% complete
Pulled 2/7 layers, 30% complete
Pulled 3/7 layers, 52% complete
Pulled 4/7 layers, 65% complete
Pulled 5/7 layers, 73% complete
Pulled 6/7 layers, 88% complete
Pulled 7/7 layers, 100% complete
Extracting
error: build error: no FROM statement found
```

Tests are working 👍

**Comment by tbugfinder** [comment]
> Hi guys, I just notice the tests picked up the first PR error:
> 
> ```
> Cloning "http://172.17.0.1:8080/scm/opendevstack/ods-core.git" ...
> 	Commit:	53bfca04109115fde9ba8d32b48980e197cc6e0b (Merge 017e9801635268c47af53ed449ae0887ad6831e8 into c89cc4c33ac466ed848b390cfa..)
> 	Author:	tbugfinder <github@online.ms>
> 	Date:	Sun Jan 12 20:39:28 2020 +0000
> Replaced Dockerfile FROM image ${JENKINS_AGENT_BASE_FROM_IMAGE}
> 
> Pulling image openshift/jenkins-slave-base-centos7 ...
> Pulled 1/7 layers, 17% complete
> Pulled 2/7 layers, 30% complete
> Pulled 3/7 layers, 52% complete
> Pulled 4/7 layers, 65% complete
> Pulled 5/7 layers, 73% complete
> Pulled 6/7 layers, 88% complete
> Pulled 7/7 layers, 100% complete
> Extracting
> error: build error: no FROM statement found
> ```
> 
> Tests are working 👍

Nice.
@hugowschneider  Which docker version is running in the environment?

**Comment by michaelsauter** [comment]
@tbugfinder This is inside OpenShift, so whatever version 3.11 uses (I think the Docker version there is quite old)

**Comment by tbugfinder** [comment]
Well, 3 years old docker 1.13.1 (2017-02-08) doesn't support ARG within FROM. So using FROM or a tag based on an input variable isn't feasible. Still the root issue is, that tag `:latest` isn't a valid image tag for a v3.11 installation.

Ref:
- https://docs.openshift.com/container-platform/3.11/release_notes/ocp_3_11_release_notes.html
- https://docs.docker.com/release-notes/docker-engine/

**Comment by tbugfinder** [comment]
I'm closing the PR as the version of the underlying docker installation is too old to support the required features.

**Pull request closed without merging**